src/index.js

```diff
@@ -208,7 +208,7 @@ export default class ImageTool {
        * Paste URL of image into the Editor
        */
       patterns: {
-        image: /https?:\/\/\S+\.(gif|jpe?g|tiff|png)$/i
+        image: /https?:\/\/\S+\.(gif|jpe?g|tiff|png)?(\?.*)$/i
```

Comment:
The `?` should be at the end of the second capturing group as a quantifier rather than matching the exact `?`. This will fail if the url doesn't have any query params.

This `https?:\/\/\S+\.(gif|jpe?g|tiff|png)(\?.*)?$`  should do the match for the urls with and without query parameters.